GoogleForCreators/web-stories-wp
# Fix PHP-CSS-Parser patching

## Summary

Fixes broken CSS gradients.

## Relevant Technical Choices

Originally, I simply wanted to use `"enable-patching": true`, but that doesn't work because of https://github.com/cweagans/composer-patches/issues/315

(cc @westonruter maybe the Composer patches in the plugin could use patches from GitHub instead of locally provided ones?)

Then, I wanted to do `"enable-patching": false` and simply provide `"patches": { ... }` myself, but that didn't work - `enable-patching` is ignored when patches are provided.

So the correct way is to provide `patches-ignore` to ignore the AMP plugin's patches in favor of our own patches declaration.

Next, I learned that `patches/php-css-parser-pull-185.patch` cannot be applied for some reason (`Could not apply patch! Skipping. The error was: Cannot apply patch patches/php-css-parser-pull-185.patch`).

Since that patch doesn't seem critical for us, I simply removed it from the patches list.

**Aside:**

I wanted to use version 1.7 of the `cweagans/composer-patches` as it has a different structure and Composer 2 support, but the AMP plugin requires 1.6.x, so that doesn't work.


## To-do



## User-facing changes

Gradients are back!

## Testing Instructions

1. Apply gradient to element
1. Notice that gradient no longer disappears on frontend.


---



Fixes #5081
